src/core/csv/reader_parsers.cc

```diff
@@ -216,9 +218,15 @@ void parse_intNN_grouped(FreadTokenizer& ctx) {
   if ((sf? sf < max_digits : ch > start) ||
       (sf == max_digits && acc <= max_value))
   {
-    T x = static_cast<T>(acc);
-    if (sizeof(T) == 4) ctx.target->int32 = negative? -x : x;
-    if (sizeof(T) == 8) ctx.target->int64 = negative? -x : x;
+    if (sizeof(T) == 4) {
+      int32_t x = static_cast<int32_t>(acc);
+      ctx.target->int32 = negative? -x : x;
+    }
+    if (sizeof(T) == 8) {
+      int64_t x = static_cast<int64_t>(acc);
+      ctx.target->int64 = negative? -x : x;
+    }
```

Comment:
Can use the same approach as in line 150 above

Reply:
The old code actually casted `x` to `T` here, but when `T` is `int64` msvc still think it is possible we assign `x` to `ctx.target->int32` (even though we never would), and issues a warning.